internal/resolution/datasource/depsdev_api.go

```diff
@@ -0,0 +1,129 @@
+package datasource
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	"sync"
+	"time"
+
+	pb "deps.dev/api/v3alpha"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+// DepsDevAPIClient is a wrapper for InsightsClient that caches requests.
+type DepsDevAPIClient struct {
+	pb.InsightsClient
+
+	// cache fields
+	mu                sync.Mutex
+	cacheTimestamp    *time.Time
+	packageCache      map[packageKey]*pb.Package
```

Comment:
Is it possible to `sync.Map` instead of locking and unlocking here?

Reply:
I *could*, it's just annoying having to cast from `any` every time when using it.
Plus it's probably better to have a single lock for the `GobEncode` / `GobDecode` functions

Reply:
Fair enough, let's stick with lock then!